### PR TITLE
Add a parameter for proxy support

### PR DESCRIPTION
**Pull request by ShivangiReja**
This pr adds one parameter to pass over proxy information when we create an AMQP connection over a web socket.

### CODE REVIEW

File: lib/connectionConfig/connectionConfig.ts [hunk 1]
```diff
@@ -80,6 +80,11 @@ export interface ConnectionConfig {
    * connection over WebSockets.
    */
   webSocketEndpointPath?: string;
+  /***
+     * @property {any} {socketOptions} - Options to be passed to the function returned by
+     * rhea.websocket_connect()
+     */
+  socketOptions?: any;
```
Comment (bterlson) [inline lib/connectionConfig/connectionConfig.ts]: Do we want to track the work for making this typed based on the constructor passed in as the webSocket configuration key?

Comment (ramya-rao-a) [reply lib/connectionConfig/connectionConfig.ts]: Yes, please create an issue and link it to the epic https://github.com/Azure/azure-sdk-for-js/issues/1056

Comment (ShivangiReja) [reply lib/connectionConfig/connectionConfig.ts]: Once we will get response back from the user, I'll create a separate PR for that.

**Comment by ShivangiReja** [comment]
@ramya-rao-a  I was thinking to give this parameter name as "webSocketOptions" but there was already a parameter "webSocketOptions" inside "connectionOptions". That's why I used only "socket".

File: lib/connectionConfig/connectionConfig.ts (hunk 1 above)
Comment (ShivangiReja) [reply lib/connectionConfig/connectionConfig.ts]: @ramya-rao-a Yes, I'll create an issue.

**Comment by ramya-rao-a** [comment]
webSocketConstructorOptions?

**Comment by ShivangiReja** [comment]
@ramya-rao-a What do you think about "webSocketInstanceOptions"?

**Comment by bterlson** [comment]
I prefer webSocketConstructorOptions to InstanceOptions fwiw (says more directly where the options get passed to).

**Comment by ramya-rao-a** [comment]
@ShivangiReja As discussed offline, close this PR and avoid making changes to the proxy branch as it is being used for testing.

Create a new branch from master with the planned changes instead

**Pull request closed without merging**